tools/cli/internal/apiversion/version.go

```diff
@@ -103,6 +104,22 @@ func WithContent(contentType string) Option {
 	}
 }
 
+// WithFullContent returns an Option to generate a new APIVersion given the contentType and contentValue.
+func WithFullContent(contentType string, contentValue *openapi3.MediaType) Option {
```

Comment:
core change: we need to consider contentValue to gather version data, since for preview and private preview we have the same content type atlas.vnd.preview+json, we can only differentiate them based on the `x-xgen-preview` extension